src/en/charms-constraints.md

```diff
@@ -1,187 +1,233 @@
-Title: Constraints
+Title: Using constraints
+TODO:  Important: include default resources requested for non-constrained machine
+       bug tracking: https://bugs.launchpad.net/juju/+bug/1768308
 
-# Constraints
+# Using constraints
 
-Constraints allow you to choose the hardware (or virtual hardware)
-to which your applications will be deployed, e.g. by specifying the amount of 
-RAM you want them to have. This is particularly useful for making sure that the
-application is deployed somewhere it can actually run efficiently, or that it is
-connected to the right network. Constraints may be set for models and
-applications, with application constraints taking precedence. Changes to
-constraints do not affect any units which have already been placed on machines.
+A *constraint* is a user-defined minimum hardware specification for a machine
+that is spawned by Juju. There are a total of nine constraints, with the most
+common ones being 'mem', 'cores', 'root-disk', and 'arch'. The definitive
+constraint resource is found on the
+[Reference: Juju constraints][reference-constraints] page.
 
-For more granularity, it is also possible to add a machine with specific 
-constraints (`juju add-machine`) and then specify that machine when deploying 
-applications ([see the documentation on `juju deploy`](./charms-deploying.html)).
+The idealized use case is that of stipulating a constraint when deploying an
+application and the backing cloud providing a machine with those exact
+resources. In the majority of cases, however, default constraints may have been
+set (at various levels) and the cloud is unable to supply those exact
+resources.
```

Comment:
I don't think it is good to split up the 'ideal' case and the explanation of what happens, which is somewhat below here.

Reply:
:ok: